docs/proposals/pepxxx_wheel_variant_support.md

```diff
@@ -62,88 +62,668 @@ something they [indirectly do today](https://github.com/wheelnext/wheelnext/pull
-    - Run once and cache
-    - Void at restart
-    - Void at plugin update
-    - Void manually: `[uv] pip cache --void variant_cache`
 
 ## Backward Compatibility
 
-The introduction of `Wheel Variants` does not break existing `installer` (`pip`, `uv`, etc.) versions, as older versions
-will simply ignore variant wheels. This is ensured by modifying the standard wheel filename regex ensures that legacy
-`pip` versions do not mistakenly install incompatible variants.
+The proposal attempts to avoid causing backwards compatibility issues by aiming for pre-variant installer
+implementations to reject variant wheels as incompatible. This is achieved by adding an additional filename component,
+causing the validation or parsing logic to fail. These installers should therefore ignore wheel variants, and fall back
+to a regular wheel, should one be provided.
+
+Aside from this explicit incompatibility, the specification makes minimal and nonintrusive changes to the wheel format.
+This aims to ease implementing it, and particularly providing support in third party tools that are neither installers
+nor build backends, and therefore do not need specific variant awareness. Variant information is stored in a separate
+file, therefore it does not require Core Metadata version change, and it is unlikely to cause problems for tools reading
+`METADATA` or `WHEEL` files.
+
+The use of explicit `*-variants.json` files make it possible to publish variant wheels via indexes without explicit
+wheel variant support. The only requirement is that the index permits publishing wheels with variant-specific filenames,
+and that it permits publishing JSON files. In particular, they do not have to parse variant information in any way.
+
 
 ## Security Implications
 
-The `Provider Plugin` mechanism introduces potential security risks, such as untrusted plugins providing misleading
-platform data. To mitigate this:
+The provider plugin mechanism introduces potential security risks. Its flexibility implies that any requested package
+may introduce variant wheels, therefore causing the package manager to install arbitrary packages for the provider
+plugin, and execute code from them. This effectively goes against [the rationale for the wheel
+format](https://peps.python.org/pep-0427/#rationale), that aimed to avoid running arbitrary code at install time.
+Furthermore, it has been pointed out that packages that will be used as a regular user are installed with elevated
+privileges, effectively making arbitrary code execution even more serious of a threat.
+
+The proposal opens at least two new points in the installation supply chain for malicious actors to inject arbitrary
+code payload:
+
+1. A new version of an existing provider plugin is published with malicious code.
+
+2. A new variant is added to an existing package, with a malicious provider.
+
+Unfortunately, it is impossible to fully address these concerns without sacrificing the flexibility and user
```

Comment:
Additional item:

- Package managers can vendor or re-implement popular plugins, such as GPU detection or CPU feature detection. This allows installations with variant support without running third-party code. Such functionality is similar to the platform probing installers already have to perform to determine operating system version and libc.